### DISCUSSION TITLE
Question regarding pre compiled packages license

### DISCUSSION THREAD

**Discussion by MohammedKHC** [termux/termux-packages#21085] [Q&A] [answered]
I need to ship cargo (rust package manager) with my app, cargo is licensed under MIT and, I know that your build script of cargo is licensed under MIT, but I am not sure about the license of pre compiled binaries that I can download from packages.termux.dev

First of all I know that it has /data/data/com.termux/files as the run path

So I used patchelf utility to replace it with my  application id.

It work great.

My question is:
Is this cargo binary licensed under MIT or other license?

((The reason I can't compile cargo and rust from the beginning for my app is that I have very bad and limited internet so I can't download the whole llvm project))

**Comment by TomJo2000** [comment, marked as answer]
Every package available on Termux contains its applicable license files.
You can find the license identifiers as part of the package `build.sh` file.
e.g.
https://github.com/termux/termux-packages/blob/1eb74d04d531367c97a483902ddad1842cbb0733/packages/cargo-c/build.sh#L3
https://github.com/termux/termux-packages/blob/1eb74d04d531367c97a483902ddad1842cbb0733/x11-packages/gtkwave/build.sh#L3
https://github.com/termux/termux-packages/blob/1eb74d04d531367c97a483902ddad1842cbb0733/packages/libmd/build.sh#L3-L4

Licenses are generally divided into 2 categories, "generic" and "specific".
- "Generic" licenses are symlinked in from a single copy contained in the `termux-licenses` package.
https://github.com/termux/termux-packages/tree/master/packages/termux-licenses/LICENSES
They do not contain project/author specific clauses or information.

- "Specific" licenses are all licenses that contain project or author specific information or clauses.
For example the MIT License, which lists the copyright year and author in the first line.
These are copied from the source files into the final package.

All licenses are placed in:
`${PREFIX}/share/doc/${TERMUX_PKG_NAME}/copyright.${counter}`
<h5>That would be<br>
<code>/data/data/com.termux/files/usr/share/doc/${TERMUX_PKG_NAME}/copyright</code><br>
on a standard Termux installation.</h5>

<h1><em></em></h1> 

So for example, `gtkwave` has `GPL-2.0` and `MIT` specified.
![image](https://github.com/user-attachments/assets/8b2fb1b5-525d-4e8b-a587-d44688bf16b6)
- GPL-2.0 is generic, so we link it in from `termux-licenses`
- MIT is specific, so we copy in every license file found in the source files.
